client/src/components/ResourceList/index.js

```diff
@@ -1,29 +1,101 @@
 import PropTypes from "prop-types";
+import { useEffect, useState } from "react";
 
+import { FormControls } from "../../components";
+import { TopicService, useService, ResourceService } from "../../services";
 import "./ResourceList.scss";
 
-export default function ResourceList({ publish, resources }) {
+export default function ResourceList({ publish, resources, pathname }) {
```

Comment:
❌ This has gone from a fairly simple presentational component, driven entirely by two props, to a complex, stateful one. I can't see why that's necessary. A few things that occur to me:

- We call `resourceService.getPublished({})` again every time we need to "fetch resources by topic", but:
- It actually _doesn't_ fetch them by topic, just re-fetches the array we were already looking at (i.e. what's already in the prop named `resources`); then
- Filters them on the client side (guaranteeing we'll see _at most_ one page's worth, and most likely much less)

We could get the same behaviour with something like:

```javascript
const displayResources = useMemo(() => {
  return selectedTopic
    ? resources.filter(/* ... */)
    : resources;
}, [resources, selectedTopic]);
```

💡 alternatively this complexity could be hoisted out of this component entirely, and dealt with by the parent _providing_ the `resources` prop. That way we wouldn't need to toggle whether or not filtering should be allowed, it wouldn't be the responsibility of this component at all.

Reply:
Actually I realise it's not the same as `resources`, it's _always_ the first 20. So the suggestion I made above would be an improvement, it would interact more coherently with pagination.

Reply:
@textbook  thank you for you valuable feedback I will soon address all above comments i push it again :+1:

Reply:
done